doc/en/example/markers.rst

```diff
@@ -393,7 +393,108 @@ The ``--markers`` option always gives you a list of available markers::
     @pytest.mark.tryfirst: mark a hook implementation function such that the plugin machinery will try to call it first/as early as possible.
     
     @pytest.mark.trylast: mark a hook implementation function such that the plugin machinery will try to call it last/as late as possible.
-    
+
+.. _`creating custom markers with extra arguments`:
+
+Creating custom markers with extra arguments
+--------------------------------------------
+
+Below is the config file that will be used in the next examples::
+
+    # content of conftest.py
+    import sys
+
+    def pytest_runtest_setup(item):
+        marker = item.get_marker('my_marker')
+        if marker is not None:
+            for info in marker:
+                print('Marker info name={} args={} kwars={}'.format(info.name, info.args, info.kwargs))
+                sys.stdout.flush()
+
+A custom marker can have its argument set, i.e. ``args`` and ``kwargs`` properties, defined by either invoking it as a callable or using ``pytest.mark.MARKER_NAME.with_args``. These two methods achieve the same effect most of the time.
+
+Let's create a custom marker using the first approach::
+
+    # content of test_custom_marker_callable_1.py
+    import pytest
+
+    @pytest.mark.my_marker('foo', bar='bar')
+    def test___call__():
+        pass
+
+Now create a custom marker using the second approach::
+
+    # content of test_custom_marker_with_args1.py
+    import pytest
+
+    @pytest.mark.my_marker.with_args('foo', bar='bar')
+    def test_with_args():
+        pass
+
+Running the tests without capturing output, we can see the two custom markers contain the same marker information::
+
+    Marker info name=my_marker args=('foo',) kwars={'bar': 'bar'}
+    .Marker info name=my_marker args=('foo',) kwars={'bar': 'bar'}
+    .
+    2 passed in 0.01 seconds
+
+Using ``pytest.mark.my_marker`` as a callable is only different from using ``with_args`` when there's a callable as the single positional argument without any keyword arguments. Let's look a this code::
+
+    # content of test_custom_marker_2.py
+    import pytest
+
+    def hello_world(*args, **kwargs):
+        return 'Hello World'
+
+    custom_marker = pytest.mark.my_marker(hello_world)
+
+    @custom_marker
+    def test___call__():
+        pass
+
+    print(custom_marker.pytestmark)
+    print(test___call__)
+
+One might expect an output similar to the output of the previous example with ``hello_world`` appearing in the ``args`` tuple. In fact, the output is::
+
```

Comment:
this should use a executed example see the other files for an example of a command output